### PR TITLE
Save and load auth method details from ide store

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-base/src/main/java/com/microsoft/azure/toolkit/intellij/base/AzurePlugin.java [hunk 1]
```diff
@@ -61,25 +61,15 @@ public void runActivity(@NotNull Project project) {
...
-                config.pluginVersion(pluginVersion);
-            }
-            if (StringUtils.isBlank(config.installationId()) && InstallationIdUtils.isValidHashMac(installationId)) {
-                config.installationId(installationId);
-            }
             AzureConfigurations.getInstance().loadState(config);
         }
         final AzureConfigurations.AzureConfigurationData config = AzureConfigurations.getInstance().getState();
-        String installationID = config.installationId();
+        String installationID = InstallationIdUtils.getHashMac();
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-base/src/main/java/com/microsoft/azure/toolkit/intellij/base/AzurePlugin.java:67]: for plugin-base, installationID is not important.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/settings/AzureConfigurations.java [hunk 2]
```diff
@@ -11,19 +11,18 @@
 import com.intellij.openapi.components.Storage;
 import com.intellij.util.xmlb.XmlSerializerUtil;
 import com.intellij.util.xmlb.annotations.MapAnnotation;
-import com.microsoft.azure.toolkit.lib.Azure;
-import com.microsoft.azure.toolkit.lib.AzureConfiguration;
+import com.microsoft.azure.toolkit.ide.common.store.IIdeStore;
 import lombok.Getter;
 import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.Nullable;
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/settings/AzureConfigurations.java]: Unify nullable

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/settings/AzureConfigurations.java [hunk 3]
```diff
@@ -11,19 +11,18 @@
...
 
 import javax.annotation.Nonnull;
-import java.io.File;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 @State(name = "Azure Toolkit for IntelliJ", storages = {@Storage("azure-data.xml")})
-public class AzureConfigurations implements PersistentStateComponent<AzureConfigurations.AzureConfigurationData> {
+public class AzureConfigurations implements PersistentStateComponent<AzureConfigurations.AzureConfigurationData>, IIdeStore {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/settings/AzureConfigurations.java]: AzureConfigurations -> IntelliJStore

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureActionsListener.java [hunk 4]
```diff
@@ -130,14 +133,17 @@ public void appFrameCreated(@NotNull List<String> commandLineArgs) {
         MvpUIHelperFactory.getInstance().init(new MvpUIHelperImpl());
 
         HDInsightLoader.setHHDInsightHelper(new HDInsightHelperImpl());
+
+        AzureStoreManager.register(new DefaultMachineStore(PluginHelper.getTemplateFile("azure-data.xml")),
+                AzureConfigurations.getInstance(), IdeaSecureStore.getInstance());
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureActionsListener.java]: IdeaSecureStore -> IntelliJSecureStore

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureConfigInitializer.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+        if (StringUtils.isBlank(installId) || InstallationIdUtils.isValidHashMac(installId)) {
+            installId = InstallationIdUtils.hash(PermanentInstallationID.get());
+        }
+
+        final AzureConfiguration config = Azure.az().config();
+        config.setMachineId(installId);
+
+        // new way of saving install id
+        AzureStoreManager.getInstance().getMachineStore().setProperty(TELEMETRY,
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureConfigInitializer.java]: Remove save properties call during load `AzureConfiguration`

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureConfigInitializer.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    private static boolean isDataFileValid() {
+        final String dataFile = PluginHelper.getTemplateFile(message("dataFileName"));
+        final File file = new File(dataFile);
+        if (!file.exists()) {
+            return false;
+        }
+        try {
+            return ParserXMLUtility.parseXMLFile(dataFile) != null;
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureConfigInitializer.java]: Migrate to json if possible

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/store/DefaultMachineStore.java [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    public void save() {
+        try {
+            Document doc = new File(dataFile).exists() ? ParserXMLUtility.parseXMLFile(dataFile) : createNewDoc();
+            map.forEach((k, v) -> {
+                updatePropertyValue(doc, k, v);
+            });
+        } catch (Exception ex) {
+            throw new AzureToolkitRuntimeException("Cannot save property due to error:" + ex.getMessage());
```
Comment (Flanker32) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/store/DefaultMachineStore.java]: pass ex to RuntimeException and modify error message

File: Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java [hunk 8]
```diff
@@ -273,11 +274,16 @@ private void initAuthMethodManagerFromSettings() {
     private static AuthMethodDetails loadSettings() {
         System.out.println("loading authMethodDetails...");
         try {
-            FileStorage fs = new FileStorage(FILE_NAME_AUTH_METHOD_DETAILS, CommonSettings.getSettingsBaseDir());
-            byte[] data = fs.read();
-            String json = new String(data);
-            if (json.isEmpty()) {
-                System.out.println(FILE_NAME_AUTH_METHOD_DETAILS + " is empty");
+            String json = AzureStoreManager.getInstance().getIdeStore().getProperty(ACCOUNT, AUTH_METHOD_DETAIL, "");
```
Comment (Flanker32) [inline Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java:277]: remove the getInstance() of AzureStoreManager?

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**